apps/ui/src/components/Modal/Payment.vue

```diff
@@ -0,0 +1,294 @@
+<script setup lang="ts">
+import { formatUnits } from '@ethersproject/units';
+import { Token } from '@/composables/usePayment';
+import { BarcodePayload } from '@/composables/usePaymentFactory';
+import { _n, clone, compareAddresses } from '@/helpers/utils';
+import { getValidator } from '@/helpers/validation';
+import { ChainId } from '@/types';
+
+const FORM = {
+  quantity: 1
+};
+
+const props = withDefaults(
+  defineProps<{
+    open: boolean;
+    unitPrice: number;
+    tokens: Token[];
+    network: ChainId;
+    barcodePayload: BarcodePayload;
+    calculator?: (unitPrice: number, quantity: number) => number;
+    quantityLabel?: string;
+  }>(),
+  {
+    quantityLabel: 'Quantity',
+    calculator: (unitPrice: number, quantity: number) => {
+      return Number((unitPrice * quantity).toFixed(2));
+    }
+  }
+);
+
+const emit = defineEmits<{
+  (e: 'close');
+  (e: 'confirmed');
+}>();
+
+const { auth } = useWeb3();
+const {
+  start: startPaymentProcess,
+  goToNextStep,
+  isLastStep,
+  currentStep
+} = usePaymentFactory(props.network);
+const { isPending, assetsMap } = useBalances({
+  treasury: toRef(() => {
+    return auth.value
+      ? {
+          chainId: props.network,
+          address: auth.value.account
+        }
+      : null;
+  })
+});
+
+const selectedTokenAddress = ref<string>('');
+const showPicker = ref(false);
+const hidden = ref(false);
+const modalTransactionProgressOpen = ref(false);
+const isTermsAccepted = ref(false);
+const form = ref(clone(FORM));
```

Comment:
We use `isTermAccepted` so maybe we can use `isHidden` and `isModalTransactionProgressOpen` to match.